### ISSUE TITLE
zql function naming conventions

### ISSUE THREAD

**Issue by mccanne** [brimsec/zq#1426]
we need a group consensus around naming conventions.  The current name all are package-style but we don't have packages so that seems wrong.  I'm not sure about lower camelCase.  It seems a little arbitrary.  Also for a interactive query language where a user often type lots of stuff rapid fire, getting capitalization right seems like a mistake.  I guess I am arguing for snake case where underscores are used only for long names that are less often typed.  How about I leave this for now and we do a PR soon on whatever we decide.

**Comment by mccanne** [comment]
Remember to change grammar rule for FunctionName when we drop "." from the names.  If we ever add a package selector to functions down the road, then the "." should be parsed and not of the name.

**Comment by mccanne** [comment]
Why was this closed?  I don't think the work has been done.

**Comment by henridf** [comment]
> Why was this closed? I don't think the work has been done.

Maybe because you wrote "Issue #1426" in https://github.com/brimsec/zq/pull/1394#discussion_r499958099 ?
(Seems like confusing UX if so).

**Comment by philrz** [comment]
I can't explain it either. The GitHub behavior of automatically closing issues is only supposed to be triggered by references in PR descriptions and commit messages, but not comments. Also, whenever an issue gets closed, I consistently see a message in the issue's history event stream that says who closed it & when, and there's none here. Based on the comment exchange, I assume @mccanne reopened it, but I don't see an entry here about _that_ either. It seems like a straight-up bug. We could take it up with GitHub Support if you'd like, though I must admit my record of getting responses out of them is not great (the last problem I reported it to them, it took _months_ to get a response, and despite their apologetic tone when they got back to me, they also never actually addressed the problem. 🙄 )

If it helps put your mind at ease, we do have a backstop that would have caught this. As part of my obsessive need to verify all our Closed issues, if I'd come across an issue in the Closed column that had no linked PR or other comment indicating it had been solved elsewhere or closed as a duplicate/wontfix, I'd have started asking questions. 😉

**Comment by philrz** [comment]
I think this can be closed because #1575 started the ball rolling by adding the alternate snake case syntax (after having established consensus with the Dev team that this was desirable), and #1379 is tracking the eventual deprecation of the package name syntax. Waiting on confirmation from @mccanne before closing.
